src/SensorFactory.cc

```diff
@@ -65,31 +64,55 @@ SensorFactory::~SensorFactory()
 
 //////////////////////////////////////////////////
 std::shared_ptr<SensorPlugin> SensorFactory::LoadSensorPlugin(
-    const std::string &_filename)
+    const std::string &_type)
 {
+  auto filename = IGN_SENSORS_PLUGIN_NAME(_type);
+
   std::string fullPath =
-      this->dataPtr->systemPaths.FindSharedLibrary(_filename);
+      this->dataPtr->systemPaths.FindSharedLibrary(filename);
   if (fullPath.empty())
   {
-    ignerr << "Unable to find sensor plugin path for [" << _filename << "]\n";
+    ignerr << "Unable to find sensor plugin path for [" << filename << "]\n";
     return std::shared_ptr<SensorPlugin>();
   }
 
-  auto pluginNames = this->dataPtr->pluginLoader.LoadLibrary(fullPath);
+  plugin::Loader pluginLoader;
+  pluginLoader.SetFlags(RTLD_LAZY | RTLD_GLOBAL);
```

Comment:
I've declined the `ign-plugin` PR that adds this option, because it proved to be troublesome.

We need to update the way `ign-sensors` is being used so that we don't need to load the plugin if the sensor is already linked against the code that's using the plugin.

I'm going to leave this PR open until we figure out how to refactor `ign-sensors` to support that use case.